source/connection-monitoring-and-pooling/tests/pool-create-max-size.yml

```diff
@@ -0,0 +1,42 @@
+version: 1
+style: unit
+description: must never exceed maxPoolSize total connections
+poolOptions:
+  maxPoolSize: 3
+operations:
+  - name: checkOut
+    label: conn1
+  - name: checkOut
+  - name: checkOut
+    label: conn2
+  - name: checkIn
+    connection: conn2
+  - name: checkOut
+  - name: start
+    target: thread1
+  - name: checkOut
+    thread: thread1
+  - name: checkIn
+    connection: conn1
+  - name: waitFor
+    target: thread1
+events:
+  - type: ConnectionPoolCreated
+  - type: ConnectionCheckOutStarted
+  - type: ConnectionCreated
+  - type: ConnectionReady
```

Comment:
Having `ConnectionReady` here complicates the tests a bit; because it isn't supposed to be emitted unless the connection has successfully completed the handshake and authenticated to the server, I don't think it's possible to emit this event without doing I/O. My understanding is that these tests are intended to be run without doing any I/O, so is it worth testing `ConnectionReady` at all? I'm not sure how we could test this in Ruby using that actual code path where the event is emitted, which means the test wouldn't be that meaningful.